pylops/waveeqprocessing/wavedecomposition.py

```diff
@@ -355,20 +365,20 @@ def PressureToVelocity(
 
 
 def UpDownComposition2D(
-    nt,
-    nr,
-    dt,
-    dr,
-    rho,
-    vel,
-    nffts=(None, None),
-    critical=100.0,
-    ntaper=10,
-    scaling=1.0,
-    backend="numpy",
-    dtype="complex128",
-    name="U",
-):
+    nt: int,
+    nr: int,
+    dt: float,
+    dr: float,
+    rho: float,
+    vel: float,
+    nffts: Optional[InputDimsLike] = (None, None),
```

Comment:
No need for Optional. The type here should be `Union[InputDimsLike, Tuple[None, None])`. Same for some other functions here.